### DISCUSSION TITLE
时间格式化聚合

### DISCUSSION THREAD

**Discussion by copkg** [apache/doris#29145] [A - General / Q&A] [answered]
比如有一张表，建表语句如下:
```
CREATE TABLE IF NOT EXISTS example_db.example_tbl_agg1
(
    `user_id` LARGEINT NOT NULL COMMENT "用户id",
    `date` DATE NOT NULL COMMENT "数据灌入日期时间",
    `city` VARCHAR(20) COMMENT "用户所在城市",
    `age` SMALLINT COMMENT "用户年龄",
    `sex` TINYINT COMMENT "用户性别",
    `last_visit_date` DATETIME REPLACE DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次访问时间",
    `cost` BIGINT SUM DEFAULT "0" COMMENT "用户总消费",
    `max_dwell_time` INT MAX DEFAULT "0" COMMENT "用户最大停留时间",
    `min_dwell_time` INT MIN DEFAULT "99999" COMMENT "用户最小停留时间"
)
AGGREGATE KEY(`user_id`, `date`, `city`, `age`, `sex`)
DISTRIBUTED BY HASH(`user_id`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);
```

表中原始数据如下:

![企业微信截图_17036684088110](https://github.com/apache/doris/assets/17465324/200948ee-8b8a-4072-bbe0-2d820a775c25)

我想以字段：last_visit_date，按照日期进行聚合，比如20171001，cost为137，20171002 cost为230，应该怎么指定datetime的格式以能满足聚合的需求？

**Comment by LemonLiTree** [comment]
to_date

**Reply by copkg** [reply]
抱歉 可能我没有描述清楚我的问题，我的意思是 我要创建一个ROLLUP，数据是Y-m-d H:i:s 格式，怎么在ROLLUP中,有2列，日期和cost，日期格式为Ymd

**Reply by copkg** [reply]
when create view，occur a error this:Materialized  view does not support this [function](function:to_date),how to use a function when create Materialized   view?

**Reply by LemonLiTree** [reply]
ROLLUP 可能不行 试试 mv？

**Reply by copkg** [reply]
mv也是不支持函数的。
会报错：materialized view does not support this function

**Reply by LemonLiTree** [reply, marked as answer]
或者 加一列 只存Ymd
